app/src/test/java/ca/bc/gov/health/ircreader/SHCDecoderTest.kt

```diff
@@ -0,0 +1,179 @@
+package ca.bc.gov.health.ircreader
+
+import ca.bc.gov.health.ircreader.model.*
+import ca.bc.gov.health.ircreader.utils.SHCDecoder
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotNull
+import org.junit.Test
+
+/**
+ * [SHCDecoderTest]
+ * This class will validate SHC Data decoder.
+ *
+ * @author Pinakin Kansara
+ */
+class SHCDecoderTest {
```

Comment:
Another way you could structure the tests is with a set of resources that represent each stage of the computation, like the ones in https://spec.smarthealth.cards/examples/. Then for each set of resources, run through the computation step by step and check the result at each stage, i.e.
1. Load the QR code resource, scan it and check it matches the numeric resource
2. Decode the numeric resource, and check the matches the JWS resource
3. Decode the JWS resource, and check the payload matches the json bundle resource

Then your tests are just a for-loop through each test-resource set.

The reason I say this, is because I am working on a script to generate these QR codes, so we can build a large set of test data for both iOS and Android and check that both of them pass all the same tests.